packages/python/openverse-attribution/src/openverse_attribution/license.py

```diff
@@ -1,128 +1,278 @@
-from enum import StrEnum
+import re
+from dataclasses import dataclass
 
-from openverse_attribution.attribution import get_attribution_text
+from openverse_attribution.data.all_licenses import all_licenses
+from openverse_attribution.license_name import LicenseName
 
 
-class License(StrEnum):
-    """
-    Represent all licenses that are handled by Openverse. This uses a very loose
-    interpretation of the term "license" as it includes licenses (both active
-    and deprecated), dedications and marks.
-    """
+KNOWN_ALIASES = {
+    "zero": "cc0",
+    "mark": "pdm",
+}
 
-    # CC licenses
-    BY = "by"
-    BY_SA = "by-sa"
-    BY_NC = "by-nc"
-    BY_ND = "by-nd"
-    BY_NC_SA = "by-nc-sa"
-    BY_NC_ND = "by-nc-nd"
 
-    # Deprecated CC licenses
-    SA = "sa"
-    SAMPLING = "sampling+"
-    NC_SAMPLING = "nc-sampling+"
+@dataclass
+class License:
+    name: LicenseName
+    ver: str | None
+    jur: str | None
 
-    # Public domain dedication
-    CC0 = "cc0"
+    def __init__(
+        self,
+        slug: str,
+        version: str | None = None,
+        jurisdiction: str | None = None,
+    ):
+        """
+        Create an instance of ``License``.
 
-    # Public domain mark
-    PDM = "pdm"
-    PUBLIC_DOMAIN = "publicdomain"
+        This function validates the license, version and jurisdiction. If some
+        fields are not provided, it makes an attempt to deduce them.
 
-    def name(self, version: str | None = None) -> str:
+        The only exception is 'publicdomain' for which the version and
+        jurisdiction fields are ignored.
+
+        Use an empty string as ``jur`` to specify a generic, universal or
+        unported form of the license.
+
+        :param slug: the slug for the license, from the ``LicenseName`` enum
+        :param version: the version of the license
+        :param jurisdiction: the jurisdiction of the license
         """
-        Get the full name of the license.
-        The ``version`` parameter is disregarded and the version is always 1.0
-        for CC0, PDM and deprecated licenses. If not provided, the version is
-        omitted for all other licenses.
 
-        :param version: the version number of the license
-        :return: the full name of the license
+        # Shorten long variable names
+        ver = version
+        jur = jurisdiction
+
+        # Handle known aliases.
+        slug = KNOWN_ALIASES.get(slug, slug)
+
+        self.slug = slug
+
+        # Encapsulates internal enum validation.
+        self.name = LicenseName(slug)
+
+        self.fallback_ver = None
+        self.fallback_jur = None
+
+        if self.name is LicenseName.PUBLICDOMAIN:
+            self.ver = None
+            self.jur = None
+            return
+
+        # Validate version against known versions.
+        if ver:
+            if ver not in all_licenses.keys():
```

Comment:
Nit: These two conditions could be combined into one.

Reply:
Fixed in cf8bc73.